code/modules/reagents/chemistry/machinery/smoke_machine.dm

```diff
@@ -63,10 +67,16 @@
 		max_range += M.rating
 	max_range = max(3, max_range)
 
+
+/obj/machinery/smoke_machine/on_set_is_operational(old_value)
```

Comment:
Might wanna check if there's any machines that process when not operational, I can imagine a few of them migth have some special effects or some other behavior

Reply:
This is redefined only for a few machines that do not process when not operational.
The base proc doesn't do anynthing.